src/charm.py

```diff
@@ -175,29 +217,143 @@ def _generate_cu_config(self) -> str:
             cu_n3_interface_name=self._charm_config.n3_interface_name,
-            cu_n3_ip_address=pod_ip,
+            cu_n3_ip_address=str(self._charm_config.n3_ip_address).split("/")[0],
             amf_external_address=self._n2_requirer.amf_ip_address,
             mcc=self._charm_config.mcc,
             mnc=self._charm_config.mnc,
             sst=self._charm_config.sst,
             tac=self._charm_config.tac,
         )
 
+    def _generate_network_annotations(self) -> List[NetworkAnnotation]:
+        """Generate a list of NetworkAnnotations to be used by CU's StatefulSet.
+
+        Returns:
+            List[NetworkAnnotation]: List of NetworkAnnotations
+        """
+        return [
+            NetworkAnnotation(
+                name=f"{self._charm_config.n3_interface_name}-net",
+                interface=self._charm_config.n3_interface_name,
+            ),
+            NetworkAnnotation(
+                name=f"{self._charm_config.n2_interface_name}-net",
+                interface=self._charm_config.n2_interface_name,
+            ),
+            NetworkAnnotation(
+                name=f"{self._charm_config.f1_interface_name}-net",
+                interface=self._charm_config.f1_interface_name,
+            ),
+        ]
+
+    @staticmethod
+    def _get_base_config() -> dict:
+        return {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "capabilities": {"mac": True},
+            },
+        }
+
+    def _get_n3_nad_config(self) -> dict:
+        n3_nad_config = self._get_base_config()
+        if self._charm_config.upf_subnet and self._charm_config.n3_gateway_ip:
+            n3_nad_config.update(
+                {
+                    "routes": [
+                        {
+                            "dst": str(self._charm_config.upf_subnet),
+                            "gw": str(self._charm_config.n3_gateway_ip),
+                        },
+                    ],
+                }
+            )
+        return self._add_cni_type_to_nad_config(
+            n3_nad_config,
+            self._charm_config.n3_ip_address,
+            self._charm_config.n3_interface_name,
+            "n3-br",
+        )
+
+    def _get_f1_nad_config(self) -> dict:
+        f1_nad_config = self._get_base_config()
+        return self._add_cni_type_to_nad_config(
+            f1_nad_config,
+            self._charm_config.f1_ip_address,
+            self._charm_config.f1_interface_name,
+            "f1-br",
+        )
+
+    def _add_cni_type_to_nad_config(
+        self, nad_config: dict, address: str, master_interface: str, bridge: str
+    ) -> dict:
+        nad_config.update(
+            {
+                "addresses": [
+                    {
+                        "address": address,
+                    }
+                ],
+            }
+        )
+        cni_type = self._charm_config.cni_type
```

Comment:
Do we need this assignment?

Reply:
As it is used twice, I define this variable. It is not necessary, removed it.